packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,389 @@
+import { p256 } from "@noble/curves/p256";
+import * as hkdf from "@noble/hashes/hkdf";
+import { sha256 } from "@noble/hashes/sha256";
+import { gcm } from "@noble/ciphers/aes";
+import {
+  uint8ArrayToHexString,
+  uint8ArrayFromHexString,
+} from "@turnkey/encoding";
+import * as bs58check from "bs58check";
+
+import {
+  AES_KEY_INFO,
+  HPKE_VERSION,
+  IV_INFO,
+  LABEL_EAE_PRK,
+  LABEL_SECRET,
+  LABEL_SHARED_SECRET,
+  SUITE_ID_1,
+  SUITE_ID_2,
+} from "./constants";
+
+interface HPKEDecyptParams {
+  ciphertextBuf: Uint8Array;
+  encappedKeyBuf: Uint8Array;
+  receiverPriv: string;
+}
+
+interface KeyPair {
+  privateKey: string;
+  publicKey: Uint8Array | string;
```

Comment:
now that we only use `KeyPair` in one place, we can probably be more specific with the expected type